extensions/transferdetails/service.go

```diff
@@ -0,0 +1,272 @@
+type Service interface {
+	documents.Service
+
+	// DeriveFromUpdatePayload derives Funding from clientUpdatePayload
+	DeriveFromUpdatePayload(ctx context.Context, req UpdateTransferDetailRequest, identifier []byte) (documents.Model, error)
+
+	// DeriveFromPayload derives Funding from clientPayload
+	DeriveFromPayload(ctx context.Context, req CreateTransferDetailRequest, identifier []byte) (documents.Model, error)
+
+	// DeriveFundingResponse returns a funding in client format
+	DeriveTransferResponse(ctx context.Context, model documents.Model, transferID string) (*TransferDetailResponse, error)
+
+	// DeriveFundingListResponse returns a funding list in client format
+	DeriveTransferListResponse(ctx context.Context, model documents.Model) (*TransferDetailListResponse, error)
+}
+
+// service implements Service and handles all funding related persistence and validations
+type service struct {
+	documents.Service
+	tokenRegistry documents.TokenRegistry
+	idSrv         identity.Service
+}
+
+const (
+	transfersLabel    = "transfer_details"
+	transfersFieldKey = "transfer_details[{IDX}]."
+	idxKey            = "{IDX}"
+	transferIDLabel   = "transfer_id"
+)
+
+// DefaultService returns the default implementation of the service.
+func DefaultService(
+	srv documents.Service,
+	tokenRegistry documents.TokenRegistry,
+) Service {
+	return service{
+		Service:       srv,
+		tokenRegistry: tokenRegistry,
+	}
+}
+
+var transfersLog = logging.Logger("tranfers-api")
+
+func deriveDIDs(data *TransferDetailData) ([]identity.DID, error) {
+	var c []identity.DID
+	for _, id := range []string{data.SenderId, data.RecipientId} {
+		if id != "" {
+			did, err := identity.NewDIDFromString(id)
+			if err != nil {
+				return nil, err
+			}
+			c = append(c, did)
+		}
+	}
+
+	return c, nil
+}
+
+func (s service) DeriveFromPayload(ctx context.Context, req CreateTransferDetailRequest, identifier []byte) (documents.Model, error) {
+	model, err := s.GetCurrentVersion(ctx, identifier)
+	if err != nil {
+		transfersLog.Error(err)
+		return nil, documents.ErrDocumentNotFound
+	}
+	attributes, err := extensions.CreateAttributesList(model, *req.Data, transfersFieldKey, transfersLabel)
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := deriveDIDs(req.Data)
+	if err != nil {
+		return nil, err
+	}
+
+	err = model.AddAttributes(
+		documents.CollaboratorsAccess{
+			ReadWriteCollaborators: c,
+		},
+		true,
+		attributes...,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	validator := CreateValidator()
+	err = validator.Validate(nil, model)
+	if err != nil {
+		return nil, errors.NewTypedError(documents.ErrDocumentInvalid, err)
+	}
+
+	return model, nil
+}
+
+// DeriveFromUpdatePayload derives Funding from clientUpdatePayload
+func (s service) DeriveFromUpdatePayload(ctx context.Context, req UpdateTransferDetailRequest, identifier []byte) (documents.Model, error) {
+	model, err := s.GetCurrentVersion(ctx, identifier)
+	if err != nil {
+		transfersLog.Error(err)
```

Comment:
```suggestion
		log.Error(err)
```

Reply:
We will have only one log instance for this entire package. So calling logger should be enough IMO